py3status/modules/gpmdp.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Display currently playing song from Google Play Music Desktop Player.
+
+Configuration parameters:
+    - cache_timeout : how often we refresh this module in seconds (5s default)
+
+Requires
+    - gpmdp
+    - gpmdp-remote
+
+@author Aaron Fields https://twitter.com/spirotot
+@license BSD
+"""
+
+# import your useful libs here
+from time import time
+from subprocess import check_output
+import shlex
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+
+    cache_timeout = 5
+    format = '{info}'
+
+    def gpmdp(self, i3s_output_list, i3s_config):
+        result = u'♫ ' + self.format
+
+        if check_output(shlex.split('gpmdp-remote status')).decode('ascii', 'ignore').strip() == 'Paused':
+            result = ''
+        else:
+            cmds = ['info', 'title', 'artist', 'album', 'status', 'current',
+                    'status', 'time_total', 'time_current', 'album_art']
+
+            for cmd in cmds:
+                if str('{' + cmd + '}') in result:
+                    result = result.replace(str('{' + cmd + '}'),check_output(shlex.split('gpmdp-remote %s' % (cmd))).decode('ascii', 'ignore').strip())
```

Comment:
Regarding your dict suggestion: a double-splat? I don't know, I mean, I'll use that code if you want me to, but it's up to you. Not the way I'd write it (obviously)... this is Python. It's supposed to be easy and make sense. We could use C if we want the code to run faster, be less easy to write, and have questionable sensibility.

There's no way that using `.format()` makes a worthwhile speed difference over `.replace()` in a string of this size... This is 2016. I'm assuming that `.format()` is actually faster than multiple `.replace()` calls, although I'm not familiar with it's internals. But if this optimization is a dealbreaker for someone's machine's capabilities/performance, then I think they've got bigger problems, like running building & running Google Play Music Desktop Player...

Please just tell me what you want (with a succinct explanation of _why_, but only if you're feeling generous); neither of our solutions seems _particularly_ bad, and I don't have time to debate every line of code. I think we'd both be happier and have more time to do more worthwhile things with our lives if we didn't. If you want me to include some code that I think is completely stupid and nonsensical (this seems like an unlikely scenario), I'll let you know, and/or just not make the change/commit myself out of principal.